pkg/controller/controller_instance.go

```diff
@@ -1550,6 +1554,9 @@ func (c *controller) processProvisionFailure(instance *v1beta1.ServiceInstance,
 		err = fmt.Errorf(failedCond.Message)
 	} else {
 		clearServiceInstanceCurrentOperation(instance)
+		// Deprovisioning is not required for provisioning that has failed with an
+		// error that doesn't require orphan mitigation
+		instance.Status.DeprovisionStatus = v1beta1.ServiceInstanceDeprovisionStatusNotRequired
```

Comment:
Apparently, currently we send a deprovision request upon deletion even if the provisioning has failed with 400 (and other client errors that have no side effects on failure)